src/reanimated2/animation/delay.ts

```diff
@@ -62,7 +69,12 @@ export const withDelay = function <T extends AnimationObject>(
           const finished =
             previousAnimation.finished ||
             previousAnimation.onFrame(previousAnimation, now);
-          animation.current = previousAnimation.current;
+          // We need to prevent setting previous animation value when there was an interruption - so that it can reset properly
+          if (didResetAfterInterruption) {
+            didResetAfterInterruption = false;
+          } else {
+            animation.current = previousAnimation.current;
+          }
```

Comment:
What if we negate the logic here and replace `!didResetAfterInterruption` with `shouldResetAfterInterruption`? This way we can simplify the logic and make it easier to understand.
```suggestion
          if (shouldResetAfterInterruption) {
            animation.current = previousAnimation.current;
            shouldResetAfterInterruption = false;
          }
```